src/axom/quest/detail/MarchingCubesSingleDomain.cpp

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "axom/config.hpp"
+
+// Implementation requires Conduit.
+#ifndef AXOM_USE_CONDUIT
+  #error "MarchingCubes.cpp requires conduit"
+#endif
+#include "conduit_blueprint.hpp"
+
+#include "axom/core/execution/execution_space.hpp"
+#include "axom/quest/detail/MarchingCubesSingleDomain.hpp"
+#include "axom/quest/detail/MarchingCubesImpl.hpp"
+#include "axom/fmt.hpp"
+
+namespace axom
+{
+namespace quest
+{
+namespace detail
+{
+namespace marching_cubes
+{
+MarchingCubesSingleDomain::MarchingCubesSingleDomain(MarchingCubes& mc)
+  : m_mc(mc)
+  , m_runtimePolicy(mc.m_runtimePolicy)
+  , m_allocatorID(mc.m_allocatorID)
+  , m_dataParallelism(mc.m_dataParallelism)
+  , m_dom(nullptr)
+  , m_ndim(0)
+  , m_topologyName()
+  , m_fcnFieldName()
+  , m_fcnPath()
+  , m_maskFieldName()
+  , m_maskPath()
+{
+  return;
+}
+
+void MarchingCubesSingleDomain::setDomain(const conduit::Node& dom,
+                                          const std::string& topologyName,
+                                          const std::string& maskField)
+{
+  m_topologyName = topologyName;
+
+  SLIC_ASSERT_MSG(
+    !conduit::blueprint::mesh::is_multi_domain(dom),
+    "MarchingCubesSingleDomain is single-domain only.  Try MarchingCubes.");
```

Comment:
Just a suggestion, but please consider inserting some extra explanation: something like 

"You passed a multi-domain Blueprint mesh to MarchingCubesSingleDomain::SetDomain. MarchingCubesSingleDomain is single-domain only. You passed a multi-domain Blueprint mesh to SetDomain. Try MarchingCubes."

Adding the extra info of explicitly what caused the error often is a life-saver in diagnosing errors.

Reply:
ok